ui/app/routes/jobs/job/index.js

```diff
@@ -54,24 +49,42 @@ export default class IndexRoute extends Route.extend(WithWatchers) {
         sortProperty: 'submitTime',
         sortDescending: true,
       });
+
+      controller.resetQueryIndex({
+        id: model.get('plainId'),
+        namespace: model.get('namespace.id'),
+      });
```

Comment:
Feels extraneous, but otherwise loading a parent job, then back out, then another parent job, then back out, then the first one again results in the child jobs from the second job being present due to the blocking query run at page load still having the previous-version's index.

There is probably another way to do this, but this appears to be working as expected.